src/main/java/com/khoros/twitterapp/TwitterApp.java

```diff
@@ -15,8 +12,18 @@ public static void main(String[] args) throws Exception {
     @Override
     public void run(TwitterAppConfiguration configuration, Environment environment) {
 
-        TwitterService.getInstance().setTWFactory(configuration.twitter4jConfigurationBuild().build());
+        TwitterFactoryComponent twComponent = DaggerTwitterFactoryComponent.builder()
+                .twitterFactoryModule(
+                        new TwitterFactoryModule(
+                                // Twitter Configuration Object
+                                configuration.twitter4jConfigurationBuild().build()
```

Comment:
better implementation, leave the configuration as just getters and setters and move all of the building to the provider